.github/workflows/publish.yaml

```diff
@@ -0,0 +1,26 @@
+name: publish
+on:
+  push:
+    branches:
+      - main
+  schedule:
+    - cron:  '0 0 * * *'
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          submodules: true
+      - name: Install Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.x
+      - name: Install Python dependencies
+        run: pip install -r requirements.txt
+      - name: Install PlantUML
+        run: sudo apt-get -y install plantuml
+      - name: Deploy mkdocs to github pages
+        run: mkdocs gh-deploy --force
```

Comment:
mkdocs comes with tooling to automatically publish to github pages. In practice the rendered page is pushed to a special branch called `gh-pages`